tmf-transform.exs

```diff
@@ -0,0 +1,225 @@
+    {"0.0", :string},
+    {"#.#", :string}
+  ]
+
+  def load(input_file_path) do
+    {:ok, package} = XlsxReader.open(input_file_path, supported_custom_formats: @custom_formats)
+
+    [sheet | _] = XlsxReader.sheet_names(package)
+
+    XlsxReader.sheet(package, sheet, empty_rows: false, number_type: String)
+  end
+end
+
+defmodule TmfReferenceModel.Transformer do
+  @moduledoc """
+  For the list of rows in a Worksheet, transform it
+  into an elixir map.
+  """
+
+  @doc """
+  transform/1 takes a list of rows from the TMF Reference model spreadsheet
+  and converts it into an Elixir map.
+  """
+  def transform(rows) when is_list(rows) do
+    rows
+    |> parse_header_rows()
+    |> transform_artifacts()
+  end
+
+  def transform({:ok, rows}), do: transform(rows)
+  def transform(any), do: IO.inspect(any, label: "Error loading rows:")
+
+  defp parse_header_rows(rows) do
+    [title_row, headers_a, headers_b | rows] = rows
+
+    %{
+      metadata: title_row |> parse_metadata(),
+      headers: parse_headers(headers_a, headers_b),
+      rows: rows
+    }
+  end
+
+  defp parse_metadata([title | title_row]) do
+    ref_model_version =
+      title_row
+      |> Enum.find(&String.contains?(&1, "Version"))
+      |> String.replace("Version", "")
+      |> String.trim()
+
+    ref_model_date =
+      title_row
+      |> Enum.find(&is_struct(&1, Date))
+      |> to_string()
+
+    title = title |> String.trim()
+
+    %{
+      :_generation_timestamp => DateTime.utc_now() |> DateTime.to_string(),
+      title: title,
+      version: ref_model_version,
+      version_date: ref_model_date
+    }
+  end
+
+  defp parse_headers(headers_a, headers_b) do
+    # There are 2 header rows, we need to combine them to handle
+    # sub sections and merged cells
+
+    {_, headers} =
+      Enum.zip_reduce(
+        headers_a,
+        headers_b,
+        {"", []},
+        fn a, b, {section, acc} ->
+          {s, h} = combine_headers(a, b, section)
+
+          # add header to accumulator list
+          {s, [acc | [h |> String.replace("\r\n", " ") |> String.trim()]]}
+        end
+      )
+
+    headers
+    |> List.flatten()
+  end
+
+  def combine_headers(a, b, section)
+  def combine_headers(a, "", _), do: {"", a}
+  def combine_headers("", b, ""), do: {"", b}
+  def combine_headers("", b, section), do: {"", concat_section_header(section, b)}
+  def combine_headers(a, b, ""), do: {a, concat_section_header(a, b)}
+
+  defp concat_section_header(section, header), do: Enum.join([section, header], " : ")
+
+  defp transform_artifacts(%{metadata: metadata, headers: headers, rows: rows}) do
+    artifacts =
+      rows
+      |> Enum.reject(fn [zone | _] -> zone == "" end)
+      |> Enum.map(fn r ->
+        # Combine the List of headers with the list of data from this row
+        {_, artifact} =
+          Enum.zip(headers, r)
+          # Now convert the list of tuples into a map
+         |> Enum.map_reduce(%{}, fn {k, v}, acc ->
```

Comment:
`Enum.into/3` doesn't give access to the collectible during iteration, it only inserts. 

However in `add_section_to_artifact/3` called [below](https://github.com/synclinical/tmf_reference_model_transform/pull/3/files#diff-e68b4c26483a23da20734add9463f2473c48d54bade17f8c9be8f23355d6ece5R129), we need access to the accumulator because a nested map is being updated. 

does that make sense?